src/channel.js

```diff
@@ -0,0 +1,554 @@
+ *        Check performance downloading binary (octet-stream) vs downloading compressed (jpeg) + decoding. 
+ *        https://github.com/cornerstonejs/codecs 
+ * NOTE: channel coloring is allowed only for monochorme channels (i.e SamplesPerPixel === 1).
+ */
+  constructor(blendingInformation) {
+    this.blendingInformation = blendingInformation;
+    this.metadata = [];
+    this.rasterSource = null;
+    this.tileLayer = null;
+  }
+
+  /** Initializes the channel building the OpenLayer objects. 
+   * All the channel have to share the same geometry properties.
+   * 
+   * @param {number[]} reference Origins array
+   * @param {number[]} reference Resolutions array
+   * @param {number[]} reference GridSizes array
+   * @param {number[]} reference TileSizes array
+   * @param {number[]} reference PixelSpacings array
+   * @param {object} OpenLayer projection
+   * @param {object} OpenLayer tileGrid
+   * @param {object} VolumeImageViewer options
+   * @param {object} VolumeImageViewer offscreen rendering engine
+   * 
+   * NOTE: in this method we set observations to the OpenLayer events 'prerender' and 'postrender',
+   *       for setting the globalCompositeOperation value to 'lighter' during the tiles blending.
+   */
+  initChannel(referenceExtent, 
+    referenceOrigins,
+    referenceResolutions,
+    referenceGridSizes,
+    referenceTileSizes,
+    referencePixelSpacings,
+    projection,
+    tileGrid,
+    options,
+    renderingEngine) {
+    
+    // cache viewer object and info in channel
+    this.blendingImageQuality = options.blendingImageQuality;  
+    this.renderingEngine = renderingEngine;
+    
+    /*
+    * To visualize images accross multiple scales, we first need to
+    * determine the image pyramid structure, i.e. the size and resolution
+    * images at the different pyramid levels.
+    */
+    
+    let geometryArrays = Channel.deriveImageGeometry(this);
+    
+    // Check that all the channels have the same pyramid parameters
+    if (arraysEqual(geometryArrays[0], referenceExtent) === false) {
+      throw new Error(
+        'Channels have different extent'
+      );
+    }
+    if (arraysEqual(geometryArrays[1], referenceOrigins) === false) {
+      throw new Error(
+        'Channels have different origins'
+      );
+    }
+    if (arraysEqual(geometryArrays[2], referenceResolutions) === false) {
+      throw new Error(
+        'Channels have different resolutions'
+      );
+    }
+    if (arraysEqual(geometryArrays[3], referenceGridSizes) === false) {
+      throw new Error(
+        'Channels have different grid sizes'
+      );
+    }
+    if (arraysEqual(geometryArrays[4], referenceTileSizes) === false) {
+      throw new Error(
+        'Channels have different tile sizes'
+      );
+    }
+    if (arraysEqual(geometryArrays[5], referencePixelSpacings) === false) {
+      throw new Error(
+        'Channels have different pixel spacings'
+      );
+    }
```

Comment:
Let's re-formulate the error messages so that they describe the issues with the DICOM images rather than the OpenLayers objects.

Reply:
The caller should not have to know anything about the OpenLayers implementation details, but should be informed about issues of the passed DICOM datasets.

Reply:
ok